cpp/libs/src/asiodnp3/DNP3Channel.cpp

```diff
@@ -136,7 +136,7 @@ template <class T>
 std::shared_ptr<T> DNP3Channel::AddStack(const LinkConfig& link, const std::shared_ptr<T>& stack)
 {
 
-	auto create = [stack, route = Route(link.RemoteAddr, link.LocalAddr), self = this->shared_from_this()]()
+	auto create = [stack, &link, this, route = Route(link.RemoteAddr, link.LocalAddr), self = this->shared_from_this()]()
```

Comment:
I don't get why this is needed. Isn't both `route` and `self` evaluated when the lambda is built? Why is it necessary to capture `link` and `this`? I'll have to re-read the standard for this...

Reply:
I would also have thought that route was fully constructed in the lambda declaration. Perhaps its a bug in this version of Clang?

Reminds me that we should add a Clang build to the CI.

Reply:
After working out the edge case that exercises this, have confirmed that it is indeed a defect in Clang 3.6 that is fixed in 3.7 :(
Relates to optimisation and templates...

Compare: https://godbolt.org/g/qCGe7K
With: https://godbolt.org/g/rM2Wx7

Reply:
Nice catch! Is LLVM 3.6 still popular in distros? I'm don't much follow LLVM project, but it's quite old (release in 2015).